CMakeLists.txt

```diff
@@ -35,8 +35,9 @@ if(NOT SWIFT_SYSTEM_NAME)
   endif()
 endif()
 
-# Don't enable WMO on Windows due to linker failures
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
+# Don't enable WMO on Windows hosts due to linker failures, and the use of swift's
+# old driver from build.ps1 when building for windows/android on a windows host.
+if(NOT CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)
```

Comment:
Can we just drop this or conditionalise it on the CMake version? CMake has had `-wmo` as the default since 3.26. The compilation mode defaults were also part of the release when CMP0157 was introduced. Effectively, this is just to enable `-wmo` with CMake 3.24..<3.26.

Reply:
Well, the problem is that WMO doesn't work very well with the old driver :(. When we build the toolchain, we don't yet have swift-driver as we need Foundation to build swift-driver. We can elide the WMO on Windows using `-D CMAKE_Swift_COMPILER_USE_OLD_DRIVER=YES` to workaround the issue for now. We will want to consider a multi-stage build so we can build with the new driver.